src/common/ceph_json.cc

```diff
@@ -249,7 +249,12 @@ bool JSONParser::parse(const char *buf_, int len)
       if (data.type() == str_type) {
         val.set(data.get_str(), true);
       } else {
-        val.set(json_spirit::write_string(data), false);
+        std::string s(json_spirit::write_string(data));
```

Comment:
> error: object backing the pointer will be destroyed at the end of the full-expression

i was impressed by the compiler's ability to spot the issue when this was `std::string_view s`

you fixed that by making a copy of the returned string, which works, but you could also use a const reference to extend the temporary's lifetime and avoid the copy. inside of the json parser, this small optimization is probably worth doing:
```c++
  const std::string& s = json_spirit::write_string(data);
```
for details on how references interact with object lifetimes, see https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary

Reply:
Done. Thanks for the link